x-pack/osquerybeat/beater/osquerybeat.go

```diff
@@ -270,22 +295,71 @@ LOOP:
 			bt.log.Infof("Exited osqueryd process, error: %v", exitErr)
 			break LOOP
 		case inputConfigs := <-inputConfigCh:
-			streams, inputTypes = config.StreamsFromInputs(inputConfigs)
-			registerActionHandlers(inputTypes)
-			setManagerPayload(inputTypes)
-			loadSchedulerStreams(streams)
+			err = handleInputConfig(inputConfigs)
+			if err != nil {
+				bt.log.Errorf("Failed to handle input configuration, err: %v, exiting", err)
+				// Cancel scheduler
+				schedCn()
+				break LOOP
+			}
 		}
 	}
 
 	// Unregister action handlers
 	unregisterActionHandlers()
 
 	// Wait for clean scheduler exit
+	bt.log.Debug("Wait clean beat run exit")
 	wg.Wait()
+	bt.log.Debug("Beat run exited, err: %v", err)
 
 	return err
 }
 
+func (bt *osquerybeat) reconnectPublisher(b *beat.Beat, inputs []config.InputConfig) (*processors.Processors, error) {
+	processors, err := processorsForInputsConfig(inputs)
+	if err != nil {
+		return nil, err
+	}
+
+	bt.log.Debugf("Connect publisher with processors: %d", len(processors.All()))
+	// Connect publisher
+	client, err := b.Publisher.ConnectWith(beat.ClientConfig{
+		Processing: beat.ProcessingConfig{
+			Processor: processors,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Swap client
+	bt.mx.Lock()
+	defer bt.mx.Unlock()
+	oldclient := bt.client
+	bt.client = client
+	if oldclient != nil {
+		oldclient.Close()
+	}
+	return processors, nil
+}
+
+func processorsForInputsConfig(inputs []config.InputConfig) (procs *processors.Processors, err error) {
+	// Use only first input processor
+	// Every input will have a processor that adds the elastic_agent info, we need only one
+	// Not expecting other processors at the moment and this needs to work for 7.13
```

Comment:
Hm, each input should run in isolation and have its own `beat.Client`. Some processors can be stateful and should not necessarily be shared.

Reply:
Currently all the inputs streams/queries are combined. The processors at the input/stream levels are not supported properly. Would require some refactoring in 7.14.